tests/integration/helpers.py

```diff
@@ -43,8 +44,15 @@ async def run_command_on_unit(unit, command: str) -> Optional[str]:
         command execution output or none if
         the command produces no output.
     """
-    action = await unit.run(command)
-    return action.results.get("Stdout", None)
+    juju_version = JujuVersion.from_environ()
+
+    # Syntax changed across Juju major versions
+    if juju_version.has_secrets:
+        action = await unit.run(command, block=True)
+        return action.results.get("stdout", None)
+    else:
+        action = await unit.run(command)
+        return action.results.get("Stdout", None)
```

Comment:
Special thanks to @shayancanonical :-)  Who went through this pain and saved me from ti :-D